In [3]:
import pandas as pd

In [5]:
file = pd.read_csv("math_students.csv")

In [7]:
file.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

In [9]:
file.iloc[::2]

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
4       GP   F   16       U     GT3       T     3     3     other     other   
6       GP   M   16       U     LE3       T     2     2     other     other   
8       GP   M   15       U     LE3       A     3     2  services     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
390     MS   M   20       U     LE3       A     2     2  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
4    ...      4        3      2     1     2      5        4   6  10  10  
6    ...      4        4      4     1     1      3        0  12  12  11  
8    ...      4        2      2     1     1      1        0  16  18  19  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
386  ...      4        4      3     2     2      5        7   6   5   6  
388  ...      4        3      4     1     1      1        0   7   9   8  
390  ...      5        5      4     4     5      4       11   9   9   9  
392  ...      5        5      3     3     3      3        3  10   8   7  
394  ...      3        2      3     3     3      5        5   8   9   9  

[198 rows x 33 columns]

Выводим только учеников с чётным количество пропусков

In [12]:
file[(file['absences'] % 2 == 0)]

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
385     MS   F   18       R     GT3       T     2     2   at_home     other   
387     MS   F   19       R     GT3       T     2     3  services     other   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
389     MS   F   18       U     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
385  ...      5        3      3     1     3      4        2  10   9  10  
387  ...      5        4      2     1     2      5        0   7   5   0  
388  ...      4        3      4     1     1      1        0   7   9   8  
389  ...      1        1      1     1     1      5        0   6   5   0  
393  ...      4        4      1     3     4      5        0  11  12  10  

[354 rows x 33 columns]

In [15]:
biggerthanten = file[(file['absences'] > 10)]

Выводим учеников с количеством пропусков больше 10

In [17]:
biggerthanten[['G3']].mean(axis=1)

18      5.0
25      8.0
29     11.0
40     11.0
44      9.0
       ... 
342    15.0
373     5.0
379    10.0
384     5.0
390     9.0
Length: 66, dtype: float64

In [62]:
biggerthanten

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
18      GP   M   17       U     GT3       T     3     2  services  services   
25      GP   F   16       U     GT3       T     2     2  services  services   
29      GP   M   16       U     GT3       T     4     4   teacher   teacher   
40      GP   F   16       U     LE3       T     2     2     other     other   
44      GP   F   16       U     LE3       T     2     2     other   at_home   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
342     GP   M   18       U     LE3       T     3     4  services     other   
373     MS   F   17       R     GT3       T     1     2     other     other   
379     MS   F   17       R     GT3       T     3     1   at_home     other   
384     MS   M   18       R     GT3       T     4     2     other     other   
390     MS   M   20       U     LE3       A     2     2  services  services   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
18   ...      5        5      5     2     4      5       16   6   5   5  
25   ...      1        2      2     1     3      5       14   6   9   8  
29   ...      4        4      5     5     5      5       16  10  12  11  
40   ...      3        3      3     1     2      3       25   7  10  11  
44   ...      4        3      3     2     2      5       14  10  10   9  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
342  ...      4        3      3     1     3      5       11  16  15  15  
373  ...      3        5      5     1     3      1       14   6   5   5  
379  ...      4        5      4     2     3      1       17  10  10  10  
384  ...      5        4      3     4     3      3       14   6   5   5  
390  ...      5        5      4     4     5      4       11   9   9   9  

[66 rows x 33 columns]

Теперь сравним среднюю итоговую оценку по ученикам с большим числом пропусков со средней оценкой по всем ученикам

In [23]:
biggerthanten[['G3']].mean(axis=1).mean()

10.151515151515152

In [25]:
file[['G3']].mean(axis=1).mean()

10.415189873417722

10.151515151515152 < 10.415189873417722 т.е. те кто много пропускал учатся хуже

In [64]:
file

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

In [88]:
female = file[(file['sex'] == 'F')]

In [90]:
female

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
385     MS   F   18       R     GT3       T     2     2   at_home     other   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
387     MS   F   19       R     GT3       T     2     3  services     other   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
389     MS   F   18       U     GT3       T     1     1     other     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
385  ...      5        3      3     1     3      4        2  10   9  10  
386  ...      4        4      3     2     2      5        7   6   5   6  
387  ...      5        4      2     1     2      5        0   7   5   0  
388  ...      4        3      4     1     1      1        0   7   9   8  
389  ...      1        1      1     1     1      5        0   6   5   0  

[208 rows x 33 columns]

In [92]:
male = file[(file['sex'] == 'M')]

In [94]:
male

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
5       GP   M   16       U     LE3       T     4     3  services     other   
6       GP   M   16       U     LE3       T     2     2     other     other   
8       GP   M   15       U     LE3       A     3     2  services     other   
9       GP   M   15       U     GT3       T     3     4     other     other   
12      GP   M   15       U     LE3       T     4     4    health  services   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
5    ...      5        4      2     1     2      5       10  15  15  15  
6    ...      4        4      4     1     1      3        0  12  12  11  
8    ...      4        2      2     1     1      1        0  16  18  19  
9    ...      5        5      1     1     1      5        0  14  15  15  
12   ...      4        3      3     1     3      5        2  14  14  14  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[187 rows x 33 columns]

In [100]:
female[['G3']].mean(axis=1).mean()

9.966346153846153

In [102]:
male[['G3']].mean(axis=1).mean()

10.914438502673796

В результате сравнения средних оценок по полу оказалось, что юноши из этих двух математических школ в среднем учатся лучше девушек

In [104]:
file

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

In [106]:
extralessons = file[(file['paid'] == 'yes')]

In [108]:
extralessons

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
5       GP   M   16       U     LE3       T     4     3  services     other   
8       GP   M   15       U     LE3       A     3     2  services     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
384     MS   M   18       R     GT3       T     4     2     other     other   
385     MS   F   18       R     GT3       T     2     2   at_home     other   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
390     MS   M   20       U     LE3       A     2     2  services  services   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
5    ...      5        4      2     1     2      5       10  15  15  15  
8    ...      4        2      2     1     1      1        0  16  18  19  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
384  ...      5        4      3     4     3      3       14   6   5   5  
385  ...      5        3      3     1     3      4        2  10   9  10  
386  ...      4        4      3     2     2      5        7   6   5   6  
388  ...      4        3      4     1     1      1        0   7   9   8  
390  ...      5        5      4     4     5      4       11   9   9   9  

[181 rows x 33 columns]

In [110]:
noextralessons = file[(file['paid'] == 'no')]

In [112]:
extralessons[['G3']].mean(axis=1).mean()

10.922651933701658

In [114]:
noextralessons[['G3']].mean(axis=1).mean()

9.985981308411215

10.922651933701658 > 9.985981308411215, средняя оценка тех кто занимаются дополнительно почти на балл выше тех кто занимается сам

In [116]:
closetohome = file[(file['reason'] == 'home')]

In [118]:
closetohome

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
6       GP   M   16       U     LE3       T     2     2     other     other   
7       GP   F   17       U     GT3       A     4     4     other   teacher   
8       GP   M   15       U     LE3       A     3     2  services     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
375     MS   F   18       R     GT3       T     1     1     other     other   
378     MS   F   18       U     GT3       T     3     3     other     other   
380     MS   M   18       U     GT3       T     4     4   teacher   teacher   
382     MS   M   17       U     GT3       T     2     3     other  services   
384     MS   M   18       R     GT3       T     4     2     other     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
6    ...      4        4      4     1     1      3        0  12  12  11  
7    ...      4        1      4     1     1      1        6   6   5   6  
8    ...      4        2      2     1     1      1        0  16  18  19  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
375  ...      4        3      2     1     2      4        2   8   8  10  
378  ...      4        1      3     1     2      1        0  15  15  15  
380  ...      3        2      4     1     4      2        4  15  14  14  
382  ...      4        4      3     1     1      3        2  11  11  10  
384  ...      5        4      3     4     3      3       14   6   5   5  

[109 rows x 33 columns]

In [130]:
first = file[(file['traveltime'] == 1)]
second = file[(file['traveltime'] == 2)]
third = file[(file['traveltime'] == 3)]
fourth = file[(file['traveltime'] == 4)]

In [132]:
first

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
5       GP   M   16       U     LE3       T     4     3  services     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
387     MS   F   19       R     GT3       T     2     3  services     other   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
390     MS   M   20       U     LE3       A     2     2  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
5    ...      5        4      2     1     2      5       10  15  15  15  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
387  ...      5        4      2     1     2      5        0   7   5   0  
388  ...      4        3      4     1     1      1        0   7   9   8  
390  ...      5        5      4     4     5      4       11   9   9   9  
392  ...      5        5      3     3     3      3        3  10   8   7  
394  ...      3        2      3     3     3      5        5   8   9   9  

[257 rows x 33 columns]

In [134]:
second

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
7       GP   F   17       U     GT3       A     4     4     other   teacher   
13      GP   M   15       U     GT3       T     4     3   teacher     other   
23      GP   M   16       U     LE3       T     2     2     other     other   
31      GP   M   15       U     GT3       T     4     4  services  services   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
383     MS   M   19       R     GT3       T     1     1     other  services   
384     MS   M   18       R     GT3       T     4     2     other     other   
385     MS   F   18       R     GT3       T     2     2   at_home     other   
389     MS   F   18       U     GT3       T     1     1     other     other   
391     MS   M   17       U     LE3       T     3     1  services  services   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
7    ...      4        1      4     1     1      1        6   6   5   6  
13   ...      5        4      3     1     2      3        2  10  10  11  
23   ...      5        4      4     2     4      5        0  13  13  12  
31   ...      4        3      1     1     1      5        0  17  16  17  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
383  ...      4        3      2     1     3      5        0   6   5   0  
384  ...      5        4      3     4     3      3       14   6   5   5  
385  ...      5        3      3     1     3      4        2  10   9  10  
389  ...      1        1      1     1     1      5        0   6   5   0  
391  ...      2        4      5     3     4      2        3  14  16  16  

[107 rows x 33 columns]

In [136]:
third

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
11      GP   F   15       U     GT3       T     2     1  services     other   
17      GP   F   16       U     GT3       T     3     3     other     other   
50      GP   F   16       U     LE3       T     2     2  services  services   
65      GP   F   16       U     LE3       T     4     3   teacher  services   
118     GP   M   17       R     GT3       T     1     3     other     other   
131     GP   F   15       U     GT3       T     1     1   at_home     other   
136     GP   M   17       R     GT3       T     3     4   at_home     other   
157     GP   F   18       R     GT3       T     1     1   at_home     other   
158     GP   M   16       R     GT3       T     2     2   at_home     other   
170     GP   M   16       U     GT3       T     3     4     other     other   
197     GP   M   16       R     LE3       T     3     3   teacher     other   
238     GP   F   17       R     GT3       T     2     1   at_home  services   
245     GP   M   16       U     GT3       T     2     1     other     other   
251     GP   M   16       U     GT3       T     3     3   at_home     other   
311     GP   F   19       U     GT3       T     2     1   at_home     other   
350     MS   M   19       R     GT3       T     1     1     other  services   
353     MS   M   19       R     GT3       T     1     1     other     other   
360     MS   F   18       R     LE3       A     1     4   at_home     other   
367     MS   F   17       R     GT3       T     1     1     other  services   
369     MS   F   18       R     GT3       T     4     4     other   teacher   
371     MS   M   18       R     LE3       T     1     2   at_home  services   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
393     MS   M   18       R     LE3       T     3     2  services     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
11   ...      5        2      2     1     1      4        4  10  12  12  
17   ...      5        3      2     1     1      4        4   8  10  10  
50   ...      4        3      3     2     3      4        2  12  13  13  
65   ...      5        4      3     1     2      1        2  16  15  15  
118  ...      5        2      4     1     4      5       20   9   7   8  
131  ...      4        3      3     1     2      4        0   8   0   0  
136  ...      5        4      5     2     4      5        0  10   0   0  
157  ...      5        2      5     1     5      4        6   9   8  10  
158  ...      4        2      2     1     2      3        2  17  15  15  
170  ...      3        4      5     2     4      2        0   6   5   0  
197  ...      3        3      4     3     5      3        8   9   9  10  
238  ...      2        1      1     1     1      3        2  13  11  11  
245  ...      4        3      3     1     1      4        6  18  18  18  
251  ...      5        3      3     1     3      2        6   7  10  10  
311  ...      3        4      1     1     1      2       20  14  12  13  
350  ...      5        4      4     3     3      2        8   8   7   8  
353  ...      4        4      4     3     3      5        4   8   8   8  
360  ...      4        3      4     1     4      5        0  13  13  13  
367  ...      5        2      1     1     2      1        0   7   6   0  
369  ...      3        2      2     4     2      5       10  14  12  11  
371  ...      4        3      3     2     3      3        3  14  12  12  
386  ...      4        4      3     2     2      5        7   6   5   6  
393  ...      4        4      1     3     4      5        0  11  12  10  

[23 rows x 33 columns]

In [138]:
fourth

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
61      GP   F   16       U     GT3       T     1     1  services  services   
108     GP   M   15       R     GT3       T     4     4     other     other   
134     GP   M   15       R     GT3       T     3     4   at_home   teacher   
149     GP   M   15       U     LE3       A     2     1  services     other   
164     GP   M   17       R     LE3       T     1     1     other  services   
228     GP   M   18       U     LE3       T     2     1   at_home     other   
327     GP   M   17       R     GT3       T     2     2  services     other   
375     MS   F   18       R     GT3       T     1     1     other     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
61   ...      5        5      5     5     5      5        6  10   8  11  
108  ...      1        3      5     3     5      1        6  10  13  13  
134  ...      5        3      3     1     1      5        0   9   0   0  
149  ...      4        5      5     2     5      5        0   8   9  10  
164  ...      5        3      5     1     5      5        0   5   8   7  
228  ...      4        3      2     4     5      3       14  10   8   9  
327  ...      4        4      5     5     5      4        8  11  10  10  
375  ...      4        3      2     1     2      4        2   8   8  10  

[8 rows x 33 columns]

In [140]:
file

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

364 человека из 395(first и second) живут в получасовой доступности от школы, это свидетельствует о том что открытие общежития около школы будет неактуально


In [ ]:
female[['G3']].mean(axis=1).mean()
male[['G3']].mean(axis=1).mean()

In [150]:
grouped = file.groupby('sex')['absences'].mean()

In [152]:
grouped

sex
F    6.216346
M    5.144385
Name: absences, dtype: float64

Получилось, что в средем девушки пропускают занятия чаще юношей

In [164]:
measure = file.pivot_table(values=['Dalc', 'Walc'], index='school', aggfunc='mean', fill_value=0)

In [166]:
measure

Dalc      Walc
school                    
GP      1.444126  2.260745
MS      1.760870  2.521739

Алкоголь чаще употребяют в школе MS(Dals - будни, Walc - выходные)

In [176]:
FvsM = file.pivot_table(values=['Medu', 'Fedu'], index='school', aggfunc='mean', fill_value=0)

In [178]:
FvsM

Fedu      Medu
school                    
GP      2.553009  2.802292
MS      2.282609  2.347826

Medu - образование матери (0 - никакого, 1 - начальное образование (4 класса), 2 – от 5 до 9 классов, 3 – среднеспециальное или 4 – высшее)

Fedu - образование отца (0 - никакого, 1 - начальное образование (4 класса), 2 – от 5 до 9 классов, 3 – среднеспециальное или 4 – высшее)


Получилось что в среднем родители учеников из школы GP - более обрахованные люди

Вывод: школа GP благополученее школы MS